server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -463,13 +466,12 @@ private MailboxId createSubMailBox(MailboxId parentMailboxId, String childMailbo
             .body(createChildMailbox)
         .when()
             .post("/jmap")
-        .thenReturn();
+        .then()
+            .contentType(JSON)
+        .extract()
+            .response();
 
-        String responseStr = response.body().asString();
-        responseStr = responseStr.substring(responseStr.indexOf("created\":"));
-        responseStr = responseStr.substring(responseStr.indexOf("id\":") + 4);
-        String mailboxId = responseStr.substring(0, responseStr.indexOf(",")).replaceAll("\"", "");
-        return new TestId.Factory().fromString(mailboxId);
+        return new TestId.Factory().fromString(response.jsonPath().get("[0][1].created." + mailboxName + ".id"));
```

Comment:
The usage of TestId.Factory() is still not ok (try to run your test with Cassandra to see why)

Reply:
In this test, it should not care about it, may we comeback to `String` instead.